File: src/com/github/luomingxu/idea/action/setting/FullDirSetting.java
package com.github.luomingxu.idea.action.setting;

import com.github.luomingxu.idea.settings.IdeaEnhanceState;
import com.github.luomingxu.idea.util.NotifyUtil;

public class FullDirSetting extends SettingAction {

    @Override
    public void set(IdeaEnhanceState state) {
        state.is = !state.is;
        NotifyUtil.info(String.format("Current setting is: %s", state.is));
    }
}

File: src/com/github/luomingxu/idea/action/setting/FullDirSetting.kt
package com.github.luomingxu.idea.action.setting

import com.github.luomingxu.idea.settings.IdeaEnhanceState
import com.github.luomingxu.idea.util.NotifyUtil

class FullDirSetting : SettingAction() {
    override fun set(settings: IdeaEnhanceState?) {
        settings?.let {
            it.`is` = !it.`is`
            NotifyUtil.info("Current setting is: ${it.`is`}")
        }
    }
}
